HI @hsoh-u.  Can you please confirm that these are expected differences (from the failing test)?

```
COMPARING ioda2nc/odb_sonde_16019_all.nc
file1: /data/output/met_test_truth/ioda2nc/odb_sonde_16019_all.nc
file2: /data/output/met_test_output/ioda2nc/odb_sonde_16019_all.nc
ERROR: NetCDF headers differ:
6c6
< 	nhdr = 2 ;
---
> 	nhdr = 96 ;
8,9c8,9
< 	nhdr_typ = 1 ;
< 	nhdr_sid = 1 ;
---
> 	nhdr_typ = 23 ;
> 	nhdr_sid = 20 ;

End of script
ERROR: Differences exist in the output
Error: Process completed with exit code 1.
```

--- a/src/tools/other/ioda2nc/ioda2nc.cc
+++ b/src/tools/other/ioda2nc/ioda2nc.cc
@@ -1262,13 +1262,15 @@ bool check_missing_thresh(float value) {
 ////////////////////////////////////////////////////////////////////////
 
 ConcatString find_meta_name(string meta_key, StringArray available_names) {
-   ConcatString metadata_name = meta_key;
-   StringArray alt_names = conf_info.metadata_map[meta_key];
-
-   for(int idx =0; idx<alt_names.n(); idx++) {
-      if(available_names.has(alt_names[idx])) {
-         metadata_name = alt_names[idx];
-         break;
+   ConcatString metadata_name;
+   if (available_names.has(meta_key)) metadata_name = meta_key;
+   else {
+      StringArray alt_names = conf_info.metadata_map[meta_key];
+      for (int idx =0; idx<alt_names.n(); idx++) {
+         if (available_names.has(alt_names[idx])) {
+            metadata_name = alt_names[idx];
+            break;
+         }
       }
    }
    return metadata_name;

